web/src/main/java/com/marklogic/hub/web/web/CurrentProjectController.java

```diff
@@ -301,13 +301,22 @@ private void clearAuthenticationAttributes(HttpServletRequest request) {
     @Override
     public void onLogoutSuccess(HttpServletRequest request, HttpServletResponse response, Authentication authentication) throws IOException, ServletException {
         request.getSession().invalidate();
-        if (watcherService.hasListener(this)) {
-            watcherService.unwatch(hubConfig.getStepDefinitionsDir().toString());
-            watcherService.unwatch(hubConfig.getHubEntitiesDir().toString());
-            watcherService.unwatch(hubConfig.getHubMappingsDir().toString());
-            watcherService.unwatch(hubConfig.getFlowsDir().toString());
-            watcherService.removeListener(this);
-        }
+        disableWatcherService(hubConfig.getStepDefinitionsDir().toFile());
+        disableWatcherService(hubConfig.getHubEntitiesDir().toFile());
+        disableWatcherService(hubConfig.getHubMappingsDir().toFile());
+        disableWatcherService(hubConfig.getFlowsDir().toFile());
+        disableWatcherService(hubConfig.getStepDefinitionsDir().toFile());
+        disableWatcherService(hubProject.getCustomModulesDir().toFile());
+        watcherService.removeListener(this);
         mappingManagerService.unsetMappingValidators();
     }
+
+    private void disableWatcherService(File dir) {
+        try{
+            watcherService.unwatch(dir.toString());
```

Comment:
One way to make this a little nicer is to verify that "dir" is not null, and that dir.exists() as well. If either is not true, no need to unwatch.